inc/class-multitaxo-plugin.php

```diff
@@ -257,13 +257,16 @@ public function multisite_set_screen_option( $status, $option, $value ) {
 	 * @return void
 	 */
 	public function load_multisite_taxonomy() {
-		$taxnow = ( isset( $_GET['multisite_taxonomy'] ) ) ? sanitize_key( wp_unslash( $_GET['multisite_taxonomy'] ) ) : null; // WPCS: input var ok, CSRF ok.
+		$page = ( isset( $_GET['page'] ) ) ? sanitize_key( wp_unslash( $_GET['page'] ) ) : null; // WPCS: input var ok, CSRF ok.
 
-		if ( empty( $taxnow ) ) {
-			wp_die( esc_html__( 'Invalid multisite taxonomy.', 'multitaxo' ) );
+		$tax_slug = str_replace( 'multisite_term_list_', '', $page );
+
+		// Check that we have something.
+		if ( empty( $tax_slug ) ) {
+			wp_die( esc_html__( 'Invalid taxonomy.', 'multitaxo' ) );
 		}
 
-		$tax = get_multisite_taxonomy( $taxnow );
+		$tax = get_multisite_taxonomy( $tax_slug );
 
 		if ( ! $tax ) {
```

Comment:
We should use is_a($tax,'Multisite_Taxonomy'), also $mulsite_taxonomy seems like a more appropriate variable name.